test/supernode_client/main.go

```diff
@@ -142,6 +142,7 @@ func sizeOfIthFile(i int64) int64 {
 }
 
 func runFileAddingWorker(n *core.IpfsNode) error {
+	errs := make(chan error, math.MaxInt32)
```

Comment:
what... what is going on in this test? `MaxInt32` ?

your changes look good to me, but we should probably reevaluate what this test is doing...